esque/clients/consumer.py

```diff
@@ -1,49 +1,75 @@
 import pathlib
-from abc import ABC, abstractmethod
-from contextlib import ExitStack
-from typing import Optional, Tuple
+from abc import ABC
+from abc import abstractmethod
+from typing import Optional, Dict
+from typing import Tuple
 
 import confluent_kafka
 import pendulum
-from confluent_kafka.cimpl import Message, TopicPartition
+from confluent_kafka.cimpl import Message
+from confluent_kafka.cimpl import TopicPartition
 
 from esque.clients.schemaregistry import SchemaRegistryClient
 from esque.config import Config
-from esque.errors import (
-    MessageEmptyException,
-    raise_for_kafka_error,
-    raise_for_message,
-    translate_third_party_exceptions,
-)
-from esque.messages.avromessage import AvroFileWriter
-from esque.messages.message import FileWriter, PlainTextFileWriter
+from esque.errors import EndOfPartitionReachedException
+from esque.errors import MessageEmptyException
+from esque.errors import raise_for_kafka_error
+from esque.errors import raise_for_message
+from esque.errors import translate_third_party_exceptions
+from esque.messages.avromessage import AvroFileWriter, StdOutAvroWriter
+from esque.messages.message import FileWriter, StdOutWriter, GenericWriter
+from esque.messages.message import PlainTextFileWriter
 from esque.ruleparser.ruleengine import RuleTree
 
 
 class AbstractConsumer(ABC):
     def __init__(self, group_id: str, topic_name: str, last: bool, match: str = None):
+        self._match = match
+        self._last = last
+        self._group_id = group_id
+        self._consumer = None
+        self._config = Config().create_confluent_config()
```

Comment:
```suggestion
        self._config = {}
```
Better to do this in `_setup_config()`